Sources/Shared/UIColorExtensions.swift

```diff
@@ -37,4 +37,10 @@ extension UIColor {
     let colorInt = UInt32(Scanner(string: string).scanInt32(representation: .hexadecimal) ?? 0)
     self.init(rgb: (Int)(colorInt))
   }
+  
+  public var rgb: Int {
+    var (r, g, b): (CGFloat, CGFloat, CGFloat) = (0.0, 0.0, 0.0)
+    getRed(&r, green: &g, blue: &b, alpha: nil)
+    return (Int(r * 255.0) << 16) | (Int(g * 255.0) << 8) | Int(b * 255.0)
+  }
```

Comment:
Widgets are already using the latest Favicon code in this PR. But this PR shouldn't be for refactoring the whole app.
I've only added one variable `var rgb: Int` to the Brave-Shared code. Refactoring all that stuff for that isn't worth it for me.

Reply:
You are already changing 261 files in this PR, so that train has left the station 🤷‍♂️ Also a large portion of this is moving files into BraveShared, so moving it into their own module isn't far off. 

Both of these things though are still not the root of this comment which is that we shouldn't be adding new code to `Shared` anymore. Anyways, you probably don't even need this anymore once you deleted `FaviconAttributesContainer`

Reply:
It is needed for `try UIColor(rgb: container.decode(Int.self, forKey: .backgroundColor))` in `Favicon.swift`
I can move the extension to `BraveShared` instead of `Shared`.